bughouse_console/src/async_server_main.rs

```diff
@@ -132,6 +132,47 @@ async fn handle_connection<S: AsyncRead + AsyncWrite + Unpin + Send + Sync + 'st
     Ok(())
 }
 
+fn check_origin<T>(req: &tide::Request<T>) -> tide::Result<()> {
+    let origin = req.header(http_types::headers::ORIGIN).map_or(
+        Err(tide::Error::from_str(
+            http_types::StatusCode::Forbidden,
+            "Failed to get Origin header of the websocket request.",
+        )),
+        |origins| Ok(origins.last().as_str()),
+    )?;
+
+    // Derive the allowed origin from this request's URL.
+    // For this to work, both the websocket endpoint and originating
+    // web page need to be hosted on the same host and port.
+    // If that changes, we'll need to check the Origin header against
+    // an allow-list.
+    if req.url().origin().ascii_serialization() != origin {
+        if req.url().host() == Some(url::Host::Domain("localhost")) {
+            let host = req.header(http_types::headers::HOST).map_or(
+                Err(tide::Error::from_str(
+                    http_types::StatusCode::Forbidden,
+                    "Failed to get Host header of the localhost websocket request.",
+                )),
+                |origins| Ok(origins.last().as_str()),
+            )?;
+            if host == "localhost" || host.starts_with("localhost") {
```

Comment:
Which case does `host.starts_with("localhost")` cover?
Wouldn't the test in the current form allow "localhosthahahaactuallynot.com"?

Reply:
Whoops, that's a mistake, fixed now, PTAL.